spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubOperations.java

```diff
@@ -81,12 +81,28 @@
 	 * Send a message to Pub/Sub.
 	 * @param topic the name of an existing topic
 	 * @param payload an object that will be serialized and sent
-	 * @param headers map of String to String headers
 	 * @return the listenable future of the call
 	 */
 	<T> ListenableFuture<String> publish(String topic, T payload,
 			Map<String, String> headers) throws IOException;
 
+	/**
+	 * Send a message to Pub/Sub.
+	 * @param topic the name of an existing topic
+	 * @param payload the message String payload
+	 * @param charset charset to decode the payload
+	 * @return the listenable future of the call
+	 */
+	ListenableFuture<String> publish(String topic, String payload, Charset charset);
```

Comment:
Well, you can have a configurable `Charset` on the template for common use-cases. And if you really need a logic based on the different charsets for different messages, you just need to perform a simple `payload.getBytes(charset)` operation before calling template. 

I just fully don't see reason to overhead template API with such a simple logic just to treat Strings some other way, where a difference is only in a small charset conversion step.